bolt/lib/Profile/YAMLProfileReader.cpp

```diff
@@ -446,6 +503,56 @@ size_t YAMLProfileReader::matchWithLTOCommonName() {
   return MatchedWithLTOCommonName;
 }
 
+size_t YAMLProfileReader::matchWithCallGraph(BinaryContext &BC) {
+  if (!opts::MatchWithCallGraph)
+    return 0;
+
+  size_t MatchedWithCallGraph = 0;
+  CGMatcher.computeBFNeighborHashes(BC);
+  CGMatcher.constructYAMLFCG(YamlBP, IdToYamLBF);
+
+  // Matches YAMLBF to BFs with neighbor hashes.
+  for (yaml::bolt::BinaryFunctionProfile &YamlBF : YamlBP.Functions) {
+    if (YamlBF.Used)
+      continue;
+    auto It = CGMatcher.YamlBFAdjacencyMap.find(&YamlBF);
+    if (It == CGMatcher.YamlBFAdjacencyMap.end())
+      continue;
+    // Computes profiled function's neighbor hash.
+    std::set<yaml::bolt::BinaryFunctionProfile *> &AdjacentFunctions =
```

Comment:
why is this a set and not a `SmallSet`/`DenseSet` ?

Reply:
I'm using std::set so that I can both perform lookups and iterate through the BF profiles in order.

Reply:
Although, this did make me think to use std::find and a SmallVector instead - I anticipate these sets to generally be quite small. I'll run some small tests and see if that improves performance